rebase this on top of the kitware/cmake

Luiz-Monad opened the pull request:
This is just a rebase of that feature on top of the kitware/cmake.

ghost commented:
[![CLA assistant check](https://cla.opensource.microsoft.com/pull/badge/signed)](https://cla.opensource.microsoft.com/microsoft/CMake?pullRequest=98) <br/>All CLA requirements met.

ahuynh-nvidia committed:
FindOpenACC: Add support for NVHPC compiler

ahuynh-nvidia committed:
NVHPC: Add support for NVIDIA HPC SDK compilers based on PGI

Identify the compilers as `NVHPC` to distinguish it from the older PGI
compilers from which they evolved, and from other `NVIDIA` compilers.

Fixes: #20887

ahuynh-nvidia committed:
FindOpenMP: Add support for NVHPC compiler

craigscott-crascit committed:
ExternalProject: Improve robustness of update step

Refactor the update logic to make it easier to follow. The following
fixes/improvements are some consequences of this change:

* Absorb a confusing git checkout failure message when the failure
  is allowed and we act on that failure appropriately.
* Fix an unnecessary fetch in some scenarios when checking out a
  git hash we already have locally.
* Stash and restore any local changes even when not rebasing.
* Avoid unsafe rebasing where we are not on a branch that is
  already tracking the requested branch.
* When fetching, use --tags --force to ensure we get all the tags
  and commits leading up to them regardless of whether the tags
  are on branches or not. Also update our local tags if they move
  on the remote.

Fixes: #20677

william-r-dieter committed:
FindOpenMP: Add support for IntelLLVM

Signed-off-by: William R. Dieter <william.r.dieter@intel.com>

william-r-dieter committed:
FindBoost: Add support for IntelLLVM

Use the same code paths as the `Intel` compiler.

Signed-off-by: William R. Dieter <william.r.dieter@intel.com>

william-r-dieter committed:
IRSL: Add support for IntelLLVM

Use the same code paths as the `Intel` compiler id.

Signed-off-by: William R. Dieter <william.r.dieter@intel.com>

william-r-dieter committed:
FindMPI: Add support for IntelLLVM

Signed-off-by: William R. Dieter <william.r.dieter@intel.com>

william-r-dieter committed:
GenerateExportHeader: Recognize Intel < 12 more strictly

Match exactly the `Intel` compiler id.

Signed-off-by: William R. Dieter <william.r.dieter@intel.com>

william-r-dieter committed:
IntelLLVM: Add support for Intel LLVM-based compilers

Using a single ID 'IntelLLVM' for the suite of Intel compilers based on
the LLVM backend.  The 'IntelLLVM' ID are used for C, C++, and Fortran.
Data Parallel C++ will be handled in a separate commit.

The C and C++ definitions are based on the Clang definitions.  The Intel
LLVM-based C and C++ compilers are based on the Clang front end, so
existing Clang options are more likely to be a good match than options
for the older Intel compilers.

Fortran is based on the older Fortran front end with the LLVM backend.
It has a similar interface to the older versions, though many options
are shared with the C and C++ compilers.

Fixes: #21561
Signed-off-by: William R. Dieter <william.r.dieter@intel.com>

william-r-dieter committed:
FindBLAS: Add support for IntelLLVM

Signed-off-by: William R. Dieter <william.r.dieter@intel.com>

william-r-dieter committed:
Tests: Update RunCMake.DependencyGraph to canonicalize paths when comparing

A symbolic link in the path was causing false path mismatches.
Resolving the symlinks makes both paths identical.

Signed-off-by: William R. Dieter <william.r.dieter@intel.com>

william-r-dieter committed:
Tests: Make GCOV test more robust to symbolic links in paths

Symbolic links in the path to the cmake build directory caused some
paths to have '..' directories in the path, which are equivalent, but
not a string match for the expected path.  As a result, some tests fail.

Signed-off-by: William R. Dieter <william.r.dieter@intel.com>

william-r-dieter committed:
Tests: Fix Fortran test C function prototypes

Several extern functions were declared without return type, which results
in warnings.  The functions are for calling Fortran subroutines, so
there should not be a return value.

Signed-off-by: William R. Dieter <william.r.dieter@intel.com>

bradking committed:
cmGlobalGenerator: Clear list of install components between runs

safreed-msft committed:
Help: Add buildPresets and testPresets to JSON schema

safreed-msft committed:
cmake-presets: Add build and test presets

Fixes: #21391

bradking committed:
Apple: Fix linking to frameworks that do not exist until build time

Fixes: #21621

asit-dhal committed:
file(GENERATE): Support new line style

Fixes: #19198

bradking committed:
cmInstallGenerator: Add backtrace to all install generators

bradking committed:
cmInstallFilesGenerator: Factor out method to get list of files

bradking committed:
cmInstallDirectoryGenerator: Factor out method to get list of directories

bradking committed:
cmInstallScriptGenerator: Factor out method to get script

bradking committed:
cmInstallExportGenerator: Add method to get installed file name

asit-dhal committed:
OUTPUT_DIRECTORY: Support tgt genex in output artifact vars

Following variables now support target dependent generator expressions.
- CMAKE_RUNTIME_OUTPUT_DIRECTORY
- CMAKE_LIBRARY_OUTPUT_DIRECTORY
- CMAKE_ARCHIVE_OUTPUT_DIRECTORY

Fixes: #18055

safreed-msft committed:
Help: Add documentation for buildPresets and testPresets

safreed-msft committed:
Tests: Factor out RunCMake.CMakePresets schema validation

Make it available to similar tests.  Also fix whitespace
in generated error message.

safreed-msft committed:
Tests: Add build preset tests

bradking committed:
cmInstallGenerator: Add accessors for settings common to all installers

william-r-dieter committed:
Tests: Update Fortran tests for IntelLLVM

Update checks for the `Intel` compiler id to match `IntelLLVM` too.

Signed-off-by: William R. Dieter <william.r.dieter@intel.com>

william-r-dieter committed:
Tests: Cleanup CPack message when files are expected and none are found

Previously, when looking for exactly one file and the number of files
was not one, the test printed out an error message saying that too many
files were found, then printing the list of files.  However, 0 is not
1 and also not too many files.

To reduce confusion, this commit adds a different message when 0 files
are found and 1 is expected.

Signed-off-by: William R. Dieter <william.r.dieter@intel.com>

william-r-dieter committed:
Tests: Update CPack DEBUGINFO tests for IntelLLVM

By default icx and icpx do not add build IDs when linking, so enable
the build ID to make the test pass.

Signed-off-by: William R. Dieter <william.r.dieter@intel.com>

william-r-dieter committed:
Tests: Update Preprocess test for IntelLLVM

Work around an icx bug in command line handling of preprocessor macros.

A `\\\"` string in a macro passed to icx on the command line is not
properly parsed in compiler versions 2021.1.0 and earlier.  This problem
is expected to be fixed, though a target release has no been set yet.

Signed-off-by: William R. Dieter <william.r.dieter@intel.com>

craigscott-crascit committed:
ExternalProject: Prevent the noisy detached head messages on checkout

When checking out a specific commit by its git hash, git will output
a very noisy message about checking out a detached HEAD. This
is not particularly helpful for us here and makes the output overall
quite verbose. Add a git config setting to prevent it.

rodlima78 committed:
CPack: add CPACK_THREADS variable to control compression threads

This allows setting how many threads the compressor will use.
Currently only implemented for XZ when using system's lzma library.

Fixes: #21715

bradking committed:
cmInstallTargetGenerator: Factor out method to get list of target files

bradking committed:
cmInstallTargetGenerator: Avoid duplicating install destination in memory

bradking committed:
cmInstallTargetGenerator: Move internal helper to anonymous namespace

craigscott-crascit committed:
Tests: Prevent the noisy CMP0114 warnings in ExternalProjectUpdate test

william-r-dieter committed:
Tests: Use the CVS :local: access method for local directories

Windows absolute paths start with a drive letter, followed by the path.
Without an explicit access method specification, CVS treats the drive
letter as a host name.  This changes explicitly adds `:local:` to local
directory repository paths used in tests to avoid this confusion.

See the CVS Manual [1] for more information.

[1] https://www.gnu.org/software/trans-coord/manual/cvs/cvs.html#Repository

Signed-off-by: William R. Dieter <william.r.dieter@intel.com>

safreed-msft committed:
Tests: Update RunCMake/CommandLine BuildDir

william-r-dieter committed:
Tests: Update MSVCRuntimeLibrary.Fortran test for IntelLLVM

`ifx` needs the same command line arguments as `ifort`.

Signed-off-by: William R. Dieter <william.r.dieter@intel.com>

cristianadam committed:
PCH: Remove restrictions for REUSE_FROM signature for MSVC

Fixes: #20201

cristianadam committed:
PCH: Remove restrictions for REUSE_FROM signature for MSVC

Fixes: #20201

gruenich committed:
LexerParser: Do not use GNU Bison's yacc mode, we use Bison extensions

Silence Bison warnings regarding use of Bison extensions.

gruenich committed:
LexerParser: Manual removing code from Bison is no longer needed

This was needed to avoid warnings while compiling the sources generated
by older versions of GNU Bison, but is not with GNU Bison 3.7.4.

gruenich committed:
LexerParser: Regenerate parser files with GNU Bison 3.7.4

Adjust expected test output.

bradking committed:
Tests: Update RunCMake.FileAPI for IntelLLVM

bradking committed:
Tests: Update Module.WriteCompilerDetectionHeader for IntelLLVM

mathstuf committed:
clang-tidy: fix `bugprone-reserved-identifier` warnings

mathstuf committed:
clang-tidy: fix `bugprone-signed-char-misuse` warnings

mathstuf committed:
clang-tidy: fix `bugprone-suspicious-include` warnings

mathstuf committed:
clang-tidy: fix `performance-no-automatic-move` warnings

mathstuf committed:
clang-tidy: fix `performance-trivially-destructible` warnings

mathstuf committed:
clang-tidy: fix `readability-make-member-function-const` warnings

mathstuf committed:
clang-tidy: fix `readability-qualified-auto` warnings

mathstuf committed:
clang-tidy: fix `readability-redundant-access-specifiers` warnings

mathstuf committed:
clang-tidy: fix `readability-redundant-string-init` warnings

mathstuf committed:
clang-tidy: fix `readability-simplify-boolean-expr` warnings

mathstuf committed:
clang-tidy: fix `readability-use-anyofallof` warnings

gruenich committed:
LexerParser: Clean up includes following include-what-you-use

bradking committed:
Tests: Update Assembler test for IntelLLVM

Enable parts of the test the same way we do for `Intel`.

bradking committed:
Tests: Fix CMP0115 warnings in CompileFeatures test

Pass source file extensions explicitly.

bradking committed:
Tests: Update CompileFeatures test for IntelLLVM

We do not have granular features for this compiler.

bradking committed:
Tests: Skip old PrecompiledHeader test on IntelLLVM

This test uses hard-coded flags for a fixed set of compilers.
We have first-class PCH support implemented for IntelLLVM.

bradking committed:
Tests: Update RunCMake.GenerateExportHeader for IntelLLVM

On Windows, the expected results are like Clang's.

Commit:
IRSL: Add support for non-x86/x64 architectures such as arm64

Implement `CMAKE_MSVC_ARCH` determination for more architectures.

Fixes: #16734

william-r-dieter committed:
Tests: Skip per file PDBs in PDBDirectoryAndName for IntelLLVM

The IntelLLVM C compiler do not yet support PDBs per file with the /Fd
option.  Disable the parts of PDBDirectoryAndName that tries to generate
them so that the test will pass.

craigscott-crascit committed:
ExternalProject: Avoid scanning docs for keywords, use include_guard()

The previous implementation was scanning the documentation in
the file at runtime to determine the set of supported keywords for
each public function. This was fragile, made it difficult to restructure
the documentation and was sometimes observable in runtime
performance measurements. Change to a more conventional
approach where supported keywords are explicitly listed in the
code.

The internal _ExternalProject_SELF variable is no longer needed.
CMake now provides CMAKE_CURRENT_FUNCTION_LIST_DIR which
can be used for the same purpose and avoids having to set a
variable when the module is read. This also removes the
requirement that the module must be included by the current or a
parent scope. It is now enough that the module has been included
once somewhere before calling any of its functions. 

The above changes combined mean that the module can now use
include_guard() and avoid having to re-parse the very long file every
time.

rodlima78 committed:
libarchive: Fix lzma_stream_encoder_mt detection

The test for lzma_stream_encoder_mt was always failing
because the lzma libraries weren't being used by linker.

francois-keith committed:
CSharp: Add 'debug:portable' to v142 flag table for VS

francois-keith committed:
CSharp: Add 'warnaserror:...' to v142 flag table for VS

This allows listing the warnings that must be flagged as errors.
For example:

    target_compile_options(proj PRIVATE "/warnaserror:1998,4014")

bradking committed:
Help: Add release note for build and test presets

bradking committed:
Merge branch 'release-3.19'

bradking committed:
Merge topic 'doc-multi-config-prop' into release-3.19

1d7daa668c Help: Add Ninja Multi-Config to list in GENERATOR_IS_MULTI_CONFIG

Acked-by: Kitware Robot <kwrobot@kitware.com>
Acked-by: Kyle Edwards <kyle.edwards@kitware.com>
Merge-request: !5737

bradking committed:
Utilities/Release: Rename Linux packages to 'linux-$arch'

bradking committed:
Utilities/Release: Rename Windows packages to 'windows-$arch'

bradking committed:
cm::optional: Fix `-Wunused-parameter` warnings in comparison operators

bradking committed:
cmNinjaUtilityTargetGenerator: Remove leftover development comment

bradking committed:
Ninja Multi-Config: Restore rebuild_cache and edit_cache targets

Since commit dcf9f4d2f7 (Ninja Multi-Config: Add support for
cross-config custom commands, 2020-12-09) we were not generating these
targets.  Restore them and add a test.

Fixes: #21722

bradking committed:
Tests: Generate simpler ExternalProject sample Git repositories

The `gitrepo.tgz` repository contains a project that enables both
C and CXX languages and builds source code.  This take a long time
to repeat over and over, particularly on Visual Studio and Xcode
generators.  Add and use a script to generate a new test repository
that has a much simpler project.

Issue: #20733

william-r-dieter committed:
Tests: Skip VSGNUFortran test for ifx.

DLLEXPORT is not implemented in ifx yet, and required by this test.

safreed-msft committed:
Tests: Add test preset tests

meresmclr committed:
Help: Document that if(EXISTS) does not expand ~

Fixes: #21729

kwrobot committed:
CMake Nightly Date Stamp

william-r-dieter committed:
Tests: Disable MFC test for IntelLLVM due to #18311

clang-cl cannot deal with implicit dependencies in UTF16 files
(see #18311).  IntelLLVM inherits this behavior from Clang.  The
MFC test uses `rc` in the way described in #18311 and gets the same
error described in #18311.  Disable the test until #18311 is fixed.

Signed-off-by: William R. Dieter <william.r.dieter@intel.com>

bradking committed:
Merge topic 'reproducible-unity-id'

ee9da76972 Unity: Generate reproducible unity IDs for anonymous namespaces

Acked-by: Kitware Robot <kwrobot@kitware.com>
Merge-request: !5727

bradking committed:
Merge topic 'update-kwsys'

1bd1f96f3f Merge branch 'upstream-KWSys' into update-kwsys
4faaa079e3 KWSys 2021-01-25 (10e36d47)

Acked-by: Kitware Robot <kwrobot@kitware.com>
Acked-by: Martin Storsjö <martin@martin.st>
Merge-request: !5736

bradking committed:
Merge topic 'csharp-vs-v142-flags'

ee78391c09 CSharp: Add 'warnaserror:...' to v142 flag table for VS
e38997d85a CSharp: Add 'debug:portable' to v142 flag table for VS

Acked-by: Kitware Robot <kwrobot@kitware.com>
Merge-request: !5735

MarkaPola committed:
cmake_path: rename 'GET ... RELATIVE_PATH' to 'GET ... RELATIVE_PART'

For consistency, subcommand 'HAS_RELATIVE_PATH' is also renamed
to 'HAS_RELATIVE_PART'.

Fixes: #21385

bradking committed:
Merge topic 'irsl-msvc-arm64'

9c5cf0c1af IRSL: Add support for non-x86/x64 architectures such as arm64

Acked-by: Kitware Robot <kwrobot@kitware.com>
Merge-request: !5733

bradking committed:
Merge topic 'cm-optional-warning'

b7d4ee23d8 cm::optional: Fix `-Wunused-parameter` warnings in comparison operators

Acked-by: Kitware Robot <kwrobot@kitware.com>
Acked-by: Kyle Edwards <kyle.edwards@kitware.com>
Merge-request: !5739

bradking committed:
Merge topic 'nmc-cache-targets'

ccbedb80e4 Ninja Multi-Config: Restore rebuild_cache and edit_cache targets
6fdbe9201f cmNinjaUtilityTargetGenerator: Remove leftover development comment

Acked-by: Kitware Robot <kwrobot@kitware.com>
Merge-request: !5740

bradking committed:
Help: Clarify meaning of CMAKE_SYSTEM_PROCESSOR

On Windows the value may not match the compiler's target architecture.
Update the documentation to state this explicitly.

Issue: #15170

bradking committed:
cmake: Reduce default recursion depth when compiling with IntelLLVM

This compiler makes stack frames large enough (at least in Debug
builds) that CMake language recursion to depth 1000 overflows the
stack.  Reduce the limit for this compiler.

bradking committed:
LexerParser: Suppress -Wused-but-marked-unused warning in cmExprParser

bradking committed:
gitlab-ci: add rules infrastructure to allow non-MR jobs

bradking committed:
gitlab-ci: consolidate release and package job infrastructure

Use `linux_release` in names associated with building or testing
release-style binaries.  Add a `.linux_package` helper to adjust
the release binary build behavior for package-specific jobs.

bradking committed:
gitlab-ci: rename build:centos6-{release => x86_64}

This makes room for a future equivalent for other architectures.

bradking committed:
gitlab-ci: add build:centos7-aarch64 job

Test building our release for the `aarch64` architecture.
For now, leave this job out during merge requests.

bradking committed:
gitlab-ci: add test:debian10-aarch64-ninja job

bradking committed:
ci: update to use CMake 3.19.3

bradking committed:
ci: download cmake on aarch64

bradking committed:
ci: generalize ninja download script to support non-upstream builds

kwrobot committed:
CMake Nightly Date Stamp

ahuynh-nvidia committed:
Tests: Update tests for NVHPC compiler

Treat it just like PGI.

bradking committed:
ci: add docker spec to build ninja for linux-aarch64

bradking committed:
ci: download ninja on aarch64

bradking committed:
Merge topic 'doc-if'

e63e83d2b8 Help: Document that if(EXISTS) does not expand ~

Acked-by: Kitware Robot <kwrobot@kitware.com>
Merge-request: !5742

bradking committed:
Merge topic 'ExternalProject-vcs-tests'

195809914f Tests: Generate simpler ExternalProject sample Git repositories

Acked-by: Kitware Robot <kwrobot@kitware.com>
Acked-by: Ben Boeckel <ben.boeckel@kitware.com>
Merge-request: !5741

bradking committed:
Merge topic 'cpack-compression-threads'

bdf30bdad8 CPack: add CPACK_THREADS variable to control compression threads
bcdb5b52a0 libarchive: Fix lzma_stream_encoder_mt detection

Acked-by: Kitware Robot <kwrobot@kitware.com>
Merge-request: !5716

bradking committed:
Merge topic 'doc-system-processor'

5a6379f1fd Help: Clarify meaning of CMAKE_SYSTEM_PROCESSOR

Acked-by: Kitware Robot <kwrobot@kitware.com>
Merge-request: !5744

bradking committed:
Merge topic 'outdir-target-genex'

64c3857780 OUTPUT_DIRECTORY: Support tgt genex in output artifact vars

Acked-by: Kitware Robot <kwrobot@kitware.com>
Merge-request: !5690

bradking committed:
Merge topic 'update-bison-parser'

7c6192472b LexerParser: Suppress -Wused-but-marked-unused warning in cmExprParser
6f64300a01 LexerParser: Clean up includes following include-what-you-use
699d50e376 LexerParser: Regenerate parser files with GNU Bison 3.7.4
c95442b9b1 LexerParser: Manual removing code from Bison is no longer needed
5a8a61a798 LexerParser: Do not use GNU Bison's yacc mode, we use Bison extensions

Acked-by: Kitware Robot <kwrobot@kitware.com>
Merge-request: !5730

bradking committed:
Merge topic 'rel-package-names'

2186f8988f Utilities/Release: Rename Windows packages to 'windows-$arch'
dc9340dbc4 Utilities/Release: Rename Linux packages to 'linux-$arch'

Acked-by: Kitware Robot <kwrobot@kitware.com>
Merge-request: !5738

bradking committed:
GenerateExportHeader: Add support for NVHPC compiler

Treat it just like PGI.

bradking committed:
Merge topic 'clang-tidy-fixes'

ef935b17ab clang-tidy: fix `readability-use-anyofallof` warnings
9ac8dbbb94 clang-tidy: fix `readability-simplify-boolean-expr` warnings
ca505718f4 clang-tidy: fix `readability-redundant-string-init` warnings
b3b6ede6a1 clang-tidy: fix `readability-redundant-access-specifiers` warnings
cdfc4e3195 clang-tidy: fix `readability-qualified-auto` warnings
808b17b120 clang-tidy: fix `readability-make-member-function-const` warnings
4470eb5179 clang-tidy: fix `performance-trivially-destructible` warnings
4f396e6528 clang-tidy: fix `performance-no-automatic-move` warnings
...

Acked-by: Kitware Robot <kwrobot@kitware.com>
Merge-request: !5731

bradking committed:
Utilities: Suppress warnings in third-party code when using IntelLLVM

kwrobot committed:
CMake Nightly Date Stamp

bradking committed:
Merge topic 'ci-aarch64'

430224b6e8 gitlab-ci: add test:debian10-aarch64-ninja job
300a91d42d gitlab-ci: add build:centos7-aarch64 job
d997637fa3 gitlab-ci: add rules infrastructure to allow non-MR jobs
a54e43670d gitlab-ci: rename build:centos6-{release => x86_64}
616bf0d2f0 gitlab-ci: consolidate release and package job infrastructure
227ca689a7 ci: download cmake on aarch64
a601ed8e0e ci: download ninja on aarch64
54c2c5f889 ci: generalize ninja download script to support non-upstream builds
...

Acked-by: Kitware Robot <kwrobot@kitware.com>
Merge-request: !5745

bradking committed:
Merge topic 'nvhpc-compiler'

4de28ba4e6 Tests: Update tests for NVHPC compiler
90502d76b7 FindOpenMP: Add support for NVHPC compiler
76dee5831b FindOpenACC: Add support for NVHPC compiler
ff7bab83c2 GenerateExportHeader: Add support for NVHPC compiler
83ea49818f NVHPC: Add support for NVIDIA HPC SDK compilers based on PGI

Acked-by: Kitware Robot <kwrobot@kitware.com>
Merge-request: !5475

bradking committed:
CMake 3.19.4

bradking committed:
Merge branch 'release-3.19'

kwrobot committed:
CMake Nightly Date Stamp

cristianadam committed:
file(CONFIGURE): Fix newlines in CONTENT

Fixes: #21749

craigscott-crascit committed:
FetchContent: Invoke steps directly and avoid a separate sub-build

The cost of setting up and executing a separate sub-build to do the
download, update and patch steps required for FetchContent population
can be significant with some platforms and CMake generators. Avoid the
sub-build altogether by invoking the step scripts directly.

Previously, if no generator was set (e.g. population was being done in
script mode), a generator needed to be available on the default PATH.
Since we no longer use a sub-build, this restriction is also now gone.

Fixes: #21703

bradking committed:
Merge topic 'intel-llvm-compilers'

9b5fd04905 Tests: Disable MFC test for IntelLLVM due to #18311
ccdf7c0005 Tests: Skip VSGNUFortran test for ifx.
ddaf412dd0 Tests: Skip per file PDBs in PDBDirectoryAndName for IntelLLVM
a619b32768 Tests: Update RunCMake.GenerateExportHeader for IntelLLVM
661b04d6a6 Tests: Skip old PrecompiledHeader test on IntelLLVM
01b1e9c40d Tests: Update CompileFeatures test for IntelLLVM
f26f10184f Tests: Update Module.WriteCompilerDetectionHeader for IntelLLVM
3ceb364b1c Tests: Update RunCMake.FileAPI for IntelLLVM
...

Acked-by: Kitware Robot <kwrobot@kitware.com>
Merge-request: !5594

bradking committed:
Help: Add release note for IntelLLVM compiler support

This was accidentally left out from commit a90d2a9eed (IntelLLVM: Add
support for Intel LLVM-based compilers, 2020-11-02).

Issue: #21561

bradking committed:
Merge topic 'externalproject-update-refactor'

ac6a4d4884 ExternalProject: Improve robustness of update step
17c4c8b92b Tests: Prevent the noisy CMP0114 warnings in ExternalProjectUpdate test
1cb65e680d ExternalProject: Prevent the noisy detached head messages on checkout

Acked-by: Kitware Robot <kwrobot@kitware.com>
Merge-request: !5389

craigscott-crascit committed:
Merge topic 'cmake_path-stabilization'

76cea3dfe2 cmake_path: rename 'GET ... RELATIVE_PATH' to 'GET ... RELATIVE_PART'

Acked-by: Kitware Robot <kwrobot@kitware.com>
Merge-request: !5743

bradking committed:
Merge topic 'intel-llvm-compilers'

b9f945eccb Help: Add release note for IntelLLVM compiler support

Acked-by: Kitware Robot <kwrobot@kitware.com>
Merge-request: !5748

DenizThatMenace committed:
FindOpenSSL: Add support for version-range

william-r-dieter committed:
Tests: Accept line breaks in more places in dpkg-shlibdeps error message

The RunCPackVerifyResult-shlibdeps-with-private-lib-failure test failed
intermittently due to sometimes breaking lines in its output in different
places.  This change accepts line breaks in more places in the command
output.

Signed-off-by: william.r.dieter <william.r.dieter@intel.com>

bradking committed:
CMakeDetermineCompilerId: Recognize XCOFF executable format

bradking committed:
cmXCOFF: Add helper to parse and edit the XCOFF binary format

william-r-dieter committed:
Tests: Add host to ENV{no_proxy} when DNS lookup failure is expected

DNS lookup of an invalid host name fails with a different error in a proxy
environment than it does in an environment without proxy.  Many tools,
including curl, use the `no_proxy` environment variable to provid a list
of hosts for which proxy should not be used.

To make lookup failure consistent, add invalid host names to the
`no_proxy` environment variable in tests that attempt to look up invalid
host names.  This way the lookup will fail consistently regardless of
whether proxy is generally used or not.

Signed-off-by: william.r.dieter <william.r.dieter@intel.com>

william-r-dieter committed:
Added `SETUID` and `SETGID` to the list of accepted file permissions

Signed-off-by: william.r.dieter <william.r.dieter@intel.com>

william-r-dieter committed:
Tests: Explicitly set permissions in CPack tests to avoid perimssions errors

When the user has the setgid bit is set on the parent directory of
the build directory, the setgid bit will be propagated throughout the
build tree.  Most tests do not care about permissions as long as they
can read and write the files the need.  The CPack tests, however, validate
that permissions match an expected set, and fail with the setgid bit set.

Explicitly set permissions on directories created in the CPackTestHelpers
to clear the setgid bit.

Signed-off-by: william.r.dieter <william.r.dieter@intel.com>

Diatrus committed:
iOS: Fix compilation of CMake itself

There are no ApplicationServices on iOS.

safreed-msft committed:
Tests: Update RunCMake/CMakePresets --list-presets

kwrobot committed:
CMake Nightly Date Stamp

kwrobot committed:
CMake Nightly Date Stamp

timhutton committed:
Help: Update to VS ClangCL toolset in user-interaction example

As of VS 2019 the toolset is now called `ClangCL`.  The toolset `LLVM`
is no longer supported.

Issue: #20415

asit-dhal committed:
try_run: Allow to set working directory

Fixes: #17634

kwrobot committed:
CMake Nightly Date Stamp

bradking committed:
Merge topic 'install-minor-refactor'

203cc694dd cmInstallGenerator: Add accessors for settings common to all installers
e32818dd76 cmInstallGenerator: Add backtrace to all install generators
eea61268e6 cmInstallExportGenerator: Add method to get installed file name
1215092507 cmInstallScriptGenerator: Factor out method to get script
2452cfdeab cmInstallDirectoryGenerator: Factor out method to get list of directories
5b23d8a7b1 cmInstallFilesGenerator: Factor out method to get list of files
5990276226 cmInstallTargetGenerator: Factor out method to get list of target files
c9e87e3447 cmInstallTargetGenerator: Avoid duplicating install destination in memory
...

Acked-by: Kitware Robot <kwrobot@kitware.com>
Merge-request: !5746

bradking committed:
Merge topic 'file-configure-newline'

6e225efd8c file(CONFIGURE): Fix newlines in CONTENT

Acked-by: Kitware Robot <kwrobot@kitware.com>
Merge-request: !5747

bradking committed:
Help: Clarify release note for IntelLLVM compiler support

Update the release note added by commit b9f945eccb (Help: Add release
note for IntelLLVM compiler support, 2021-01-29) to distinguish between
the Intel oneAPI NextGen and Classic compilers.

Co-Author: William R. Dieter <william.r.dieter@intel.com>
Suggested-by: Michael Hirsch, Ph.D <michael@scivision.dev>

bradking committed:
Rename CMAKE_USE_ELF_PARSER to CMake_USE_ELF_PARSER

We use the `CMake_` prefix for options affecting CMake itself.

bradking committed:
Rename CMAKE_USE_MACH_PARSER to CMake_USE_MACH_PARSER

We use the `CMake_` prefix for options affecting CMake itself.

robertmaynard committed:
FindCUDAToolkit: Restore use of CUDA_PATH environment variable

Refactoring in commit 7cc815a2a6 (CUDAToolkit: Detect CUDA SDK that
don't have nvcc, 2020-07-24, v3.19.0-rc1~366^2) accidentally broke
use of the `CUDA_PATH` environment variable.

Fixes: #21740

kwrobot committed:
CMake Nightly Date Stamp

bradking committed:
Merge topic 'bad-dns-no_proxy'

c0b1f5f30c Tests: Add host to ENV{no_proxy} when DNS lookup failure is expected

Acked-by: Kitware Robot <kwrobot@kitware.com>
Merge-request: !5752

bradking committed:
Merge topic 'tests-setgid'

8213390a3e Added `SETUID` and `SETGID` to the list of accepted file permissions
f41d0e0c77 Tests: Explicitly set permissions in CPack tests to avoid perimssions errors

Acked-by: Kitware Robot <kwrobot@kitware.com>
Merge-request: !5754

bradking committed:
Merge topic 'shlibdeps-fixup'

1aefa86c31 Tests: Accept line breaks in more places in dpkg-shlibdeps error message

Acked-by: Kitware Robot <kwrobot@kitware.com>
Merge-request: !5751

bradking committed:
Merge topic 'fix21727_for_FindOpenSSL'

bc00cf9300 FindOpenSSL: Add support for version-range

Acked-by: Kitware Robot <kwrobot@kitware.com>
Merge-request: !5750

bradking committed:
Merge topic 'doc-user-interaction-ClangCL'

c16509907c Help: Update to VS ClangCL toolset in user-interaction example

Acked-by: Kitware Robot <kwrobot@kitware.com>
Merge-request: !5756

bradking committed:
Merge topic 'issue-19198'

255df8622b file(GENERATE): Support new line style

Acked-by: Kitware Robot <kwrobot@kitware.com>
Acked-by: Brad King <brad.king@kitware.com>
Merge-request: !5649

cristianadam committed:
file(CONFIGURE): Use text mode for default OUTPUT content

This affects only Windows where \n will be translated as \r\n

Fixes: #21769

kwrobot committed:
KWSys 2021-02-02 (c672435e)

Code extracted from:

    https://gitlab.kitware.com/utils/kwsys.git

at commit c672435eba0d5b024117cd4f30d8d2ed57f7f338 (master).

Upstream Shortlog
-----------------

Brad King (1):
      6e51fe76 FStream: Add std::fstream wrapper for in/out file streams

bradking committed:
Merge branch 'upstream-KWSys' into update-kwsys

# By KWSys Upstream
* upstream-KWSys:
  KWSys 2021-02-02 (c672435e)

bradking committed:
Merge topic 'pch-reuse-from-obj-msvc'

9f06097141 PCH: Remove restrictions for REUSE_FROM signature for MSVC

Acked-by: Kitware Robot <kwrobot@kitware.com>
Rejected-by: Cristian Adam <cristian.adam@gmail.com>
Acked-by: Cristian Adam <cristian.adam@gmail.com>
Merge-request: !5728

bradking committed:
Revert "PCH: Remove restrictions for REUSE_FROM signature for MSVC"

This reverts commit 9f060971411aca979807f70307d8b9fe1b43ff24.
It was merged accidentally.

bradking committed:
Merge topic 'revert-pch-reuse-from-obj-msvc'

8d379e7406 Revert "PCH: Remove restrictions for REUSE_FROM signature for MSVC"

Acked-by: Kitware Robot <kwrobot@kitware.com>
Merge-request: !5764

bradking committed:
Tests: Add sample XCOFF binaries

Generate the binaries as follows:

    $ cat main.c
    int main(void) { return 0; }
    $ xlc -q64 -o xcoff64.bin -Wl,-blibpath:/sample/rpath:/usr/lib:/lib main.c
    $ strip -X 64 xcoff64.bin
    $ xlc -q32 -o xcoff32.bin -Wl,-blibpath:/sample/rpath:/usr/lib:/lib main.c
    $ strip -X 32 xcoff32.bin

bradking committed:
Tests: Convert CMake.ELF to RunCMake.file-RPATH ELF case

bradking committed:
Ninja Multi-Config: Enable relink diagnostic message

We already have a message for the "Ninja" generator, but it applies
to the "Ninja Multi-Config" generator too.

bradking committed:
AIX: Enable XCOFF editing to replace RPATH on installation

Avoid relinking before installation.

kwrobot committed:
CMake Nightly Date Stamp

Commit:
curl 2021-02-03 (2f33be81)

Code extracted from:

    https://github.com/curl/curl.git

at commit 2f33be817cbce6ad7a36f27dd7ada9219f13584c (curl-7_75_0).

craigscott-crascit committed:
cmComputeLinkInformation: Fix misspelt private variable name

craigscott-crascit committed:
Xcode: Don't hard-code SDK-provided implicit framework search paths

When a framework is linked to a target by its full path and that
framework is located in one of the implicit framework search directories,
CMake 3.18.5 and earlier discarded that path.
ce2dee9e5ba (Xcode: Don't add framework as -framework argument in
linker info list, 2020-09-28) introduced a regression which resulted in
the framework path always being added to the search path even if it
matched one of the implicit search paths. This broke the ability to do
device and simulator builds from the same configured project.

Fixes: #21678

bradking committed:
Merge topic 'update-kwsys'

7d46310cf8 Merge branch 'upstream-KWSys' into update-kwsys
86ecce2072 KWSys 2021-02-02 (c672435e)

Acked-by: Kitware Robot <kwrobot@kitware.com>
Merge-request: !5763

bradking committed:
Merge topic 'cmake-compile-on-iOS'

30360cb05f iOS: Fix compilation of CMake itself

Acked-by: Kitware Robot <kwrobot@kitware.com>
Merge-request: !5753

bradking committed:
Merge topic 'file-configure-endl'

05f16ca7ee file(CONFIGURE): Use text mode for default OUTPUT content

Acked-by: Kitware Robot <kwrobot@kitware.com>
Acked-by: Asit Dhal <dhal.asitk@gmail.com>
Merge-request: !5762

bradking committed:
Merge topic 'pch-reuse-from-obj-msvc'

c450d66daa PCH: Remove restrictions for REUSE_FROM signature for MSVC

Acked-by: Kitware Robot <kwrobot@kitware.com>
Merge-request: !5765

bradking committed:
Merge topic 'findcudatoolkit_env_path'

4ea6d3c3ae FindCUDAToolkit: Restore use of CUDA_PATH environment variable

Acked-by: Kitware Robot <kwrobot@kitware.com>
Acked-by: Raul Tambre <raul@tambre.ee>
Merge-request: !5761

bradking committed:
Merge topic 'findcudatoolkit_env_path' into release-3.19

4ea6d3c3ae FindCUDAToolkit: Restore use of CUDA_PATH environment variable

Acked-by: Kitware Robot <kwrobot@kitware.com>
Acked-by: Raul Tambre <raul@tambre.ee>
Merge-request: !5761

bradking committed:
Merge branch 'release-3.19'

bradking committed:
Merge topic 'intel-llvm-compilers'

a552487843 Help: Clarify release note for IntelLLVM compiler support

Acked-by: Kitware Robot <kwrobot@kitware.com>
Merge-request: !5758

bradking committed:
curl: Update script to get curl 7.75.0

bradking committed:
Merge branch 'upstream-curl' into update-curl

* upstream-curl:
  curl 2021-02-03 (2f33be81)

bradking committed:
curl: Set build options the way we need for CMake

bradking committed:
FindBoost: Add support for Boost 1.75

Update the list of known versions.

Run the command

    cmake -DBOOST_DIR=/path/to/boost_1_75_0 \
      -P Utilities/Scripts/BoostScanDeps.cmake

to extract dependencies from the 1.75.0 source tree.
They differ from 1.74's dependencies by the addition of
dependencies of the json component, so add a new version
block to FindBoost.

Fixes: #21773

concatime committed:
bootstrap: add cmake_ld_flags to include check

robertmaynard committed:
CMakeDetermineCompilerABI: Parse library arch from versioned paths

Teach CMake how to extract `CMAKE_<LANG>_LIBRARY_ARCHITECTURE` from
versioned paths such as `/usr/lib/gcc/x86_64-linux-gnu/9`. These kind of
paths are generated by NVHPC compilers.

robertmaynard committed:
CUDA: Always extract CUDA Toolkit root from nvcc verbose output

Fixes #21750, #21763

Given that NVCC can be provided by multiple different sources (NVIDIA HPC SDK, CUDA Toolkit, distro)
each of which has a different layout, we need to extract the CUDA toolkit root from the compiler
itself, allowing us to support numerious different scattered toolkit layouts.

The NVIDIA HPC SDK specifically ships two copies of nvcc one in
`compilers/bin/` and one in `cuda/bin`. Thus when using
`compilers/bin/nvcc` the Toolkit root logic fails.

kwrobot committed:
CMake Nightly Date Stamp

craigscott-crascit committed:
add_library(): Allow imported object libraries with multi-arch

Fixes: #21276

bradking committed:
Merge topic 'xcode-framework-path' into release-3.19

5389bb4274 Xcode: Don't hard-code SDK-provided implicit framework search paths
df08f8df30 cmComputeLinkInformation: Fix misspelt private variable name
375b307bae Apple: Fix linking to frameworks that do not exist until build time

Acked-by: Kitware Robot <kwrobot@kitware.com>
Merge-request: !5760

bradking committed:
Merge branch 'release-3.19'

bradking committed:
Merge topic 'xcode-framework-path'

5389bb4274 Xcode: Don't hard-code SDK-provided implicit framework search paths
df08f8df30 cmComputeLinkInformation: Fix misspelt private variable name
375b307bae Apple: Fix linking to frameworks that do not exist until build time

Acked-by: Kitware Robot <kwrobot@kitware.com>
Merge-request: !5760

bradking committed:
Merge topic 'FindBoost-1.75' into release-3.19

8293064760 FindBoost: Add support for Boost 1.75

Acked-by: Kitware Robot <kwrobot@kitware.com>
Merge-request: !5768

bradking committed:
Merge branch 'release-3.19'

bradking committed:
Merge topic 'FindBoost-1.75'

8293064760 FindBoost: Add support for Boost 1.75

Acked-by: Kitware Robot <kwrobot@kitware.com>
Merge-request: !5768

bradking committed:
Merge topic 'update-curl'

24f3388fa4 curl: Set build options the way we need for CMake
426ef61cc8 Merge branch 'upstream-curl' into update-curl
076b3219f5 curl 2021-02-03 (2f33be81)
0341888c00 curl: Update script to get curl 7.75.0

Acked-by: Kitware Robot <kwrobot@kitware.com>
Merge-request: !5767

bradking committed:
Merge topic 'bootstrap-include-check-ldflags'

11260f6c20 bootstrap: add cmake_ld_flags to include check

Acked-by: Kitware Robot <kwrobot@kitware.com>
Merge-request: !5770

bradking committed:
Merge topic 'issue-17634'

5af38a11ce try_run: Allow to set working directory

Acked-by: Kitware Robot <kwrobot@kitware.com>
Merge-request: !5757

bradking committed:
Merge topic 'aix-xcoff-edit'

e017ba046c AIX: Enable XCOFF editing to replace RPATH on installation
56fc4a325f cmXCOFF: Add helper to parse and edit the XCOFF binary format
ddaaee907d CMakeDetermineCompilerId: Recognize XCOFF executable format
69e1d95a8a Tests: Add sample XCOFF binaries
f79d991dfd Tests: Convert CMake.ELF to RunCMake.file-RPATH ELF case
d8f3e68ca9 Ninja Multi-Config: Enable relink diagnostic message
cdcfe3eb99 Rename CMAKE_USE_MACH_PARSER to CMake_USE_MACH_PARSER
b6071c93f5 Rename CMAKE_USE_ELF_PARSER to CMake_USE_ELF_PARSER

Acked-by: Kitware Robot <kwrobot@kitware.com>
Merge-request: !5769

bradking committed:
Help: Restore docs that find_library considers LIB, not INCLUDE

Documentation updates in commit 02f527c66a (Find: Provide global
controls for the `NO_[]_PATH` call options, 2019-06-12,
v3.16.0-rc1~541^2) accidentally switched the `find_library`
documentation to mention `INCLUDE` where it should be `LIB`.

While at it, update `find_file` and `find_path` to mention
`INCLUDE` and `PATH` in the order they are considered.

craigscott-crascit committed:
ExternalProject: Refactor pre-configure steps to support no-target uses

The mkdir, download, update and patch steps are used by
FetchContent during the configure phase of the main build. Because
these steps need a target, this has so far required a sub-build to be
set up. The changes here factor out the preparation of the scripts
from the creation of the targets, allowing future work to leverage these
steps without a sub-build (see #21703).

As part of the refactoring, some rationalisation of the stamp files,
repository info files and script names was done to make things more
consistent between download methods and step implementations.
Every download method now records its own specific repository info
in a file and that file is a dependency of the download step. The source
directory is also written to that file, so if the SOURCE_DIR changes, the
download will be retriggered (the existing implementation fails in this
scenario). Each download method now also has just one driver script
that implements the whole step (it may pull in other scripts to do its
task though). The patch step gained support for USES_TERMINAL as
a result of generalising the implementation for custom commands.

Fixes: #21748

kwrobot committed:
CMake Nightly Date Stamp

craigscott-crascit committed:
FindGit: Cache the version more effectively

In 315a200f0c4 (FindGit: Cache the GIT_EXECUTABLE version for the
current run, 2021-01-20), the GIT_VERSION_STRING was meant to be
cached after the first time it was computed for a given GIT_EXECUTABLE
location. That logic assumed GIT_VERSION_STRING would be visible in
the current scope, but it might not be. The global property alone is
enough to check whether the version has been determined previously,
so don't switch the logic based on whether GIT_VERSION_STRING is
defined or not.

Relates: #21703

bradking committed:
LexerParser: Suppress gcc -Wfree-nonheap-object warnings in generated code

bradking committed:
Merge topic 'doc-find-env-vars'

c40e81ce80 Help: Restore docs that find_library considers LIB, not INCLUDE

Acked-by: Kitware Robot <kwrobot@kitware.com>
Merge-request: !5772

bradking committed:
Merge branch 'release-3.19'

bradking committed:
Merge topic 'doc-find-env-vars' into release-3.19

c40e81ce80 Help: Restore docs that find_library considers LIB, not INCLUDE

Acked-by: Kitware Robot <kwrobot@kitware.com>
Merge-request: !5772

bradking committed:
Merge topic 'build-test-presets'

9b5289ebd8 Help: Add release note for build and test presets
c8a5cd6871 Tests: Update RunCMake/CMakePresets --list-presets
69a5cf23a2 Tests: Update RunCMake/CommandLine BuildDir
b500935b94 Tests: Add test preset tests
74a86566f0 Tests: Add build preset tests
56751c83aa Tests: Factor out RunCMake.CMakePresets schema validation
676ecf0d37 cmake-presets: Add build and test presets
4f4f2028b8 Help: Add documentation for buildPresets and testPresets
...

Acked-by: Kitware Robot <kwrobot@kitware.com>
Merge-request: !5681

bradking committed:
Merge topic 'fetchcontent-performance'

17e5516e60 FetchContent: Invoke steps directly and avoid a separate sub-build
4f3d1abbb4 ExternalProject: Refactor pre-configure steps to support no-target uses
23aab9ecce ExternalProject: Avoid scanning docs for keywords, use include_guard()

Acked-by: Kitware Robot <kwrobot@kitware.com>
Acked-by: Robert Maynard <robertjmaynard@gmail.com>
Acked-by: Michael Hirsch, Ph.D. <michael@scivision.dev>
Acked-by: Lars Melchior <lars.melchior@gmail.com>
Merge-request: !5749

bradking committed:
LexerParser: Do not override existing _POSIX_C_SOURCE definition

In commit f034b0f663 (CMake compilation: do not use compiler extensions,
2020-03-14, v3.18.0-rc1~494^2) we explicitly defined `_POSIX_C_SOURCE`
to ensure availability of POSIX APIs even when compiler extensions are
not enabled.  Update the code to avoid redefining `_POSIX_C_SOURCE` if
it is already defined.  This occurs when building our release binaries
as configured in `Utilities/Release/linux`, where we define
`_POSIX_C_SOURCE` explicitly on the compiler command line.

craigscott-crascit committed:
Tests: TargetObjects does not need any languages enabled

bradking committed:
Merge branch 'lexer-posix-source' into release-3.18

Merge-request: !5776

bradking committed:
Help: Convert some genex names to explicit cross-references

Since commit c2dc7e0f53 (Help: Convert genex documentation to sphinx
domain objects, 2021-01-15) we can use `:genex:` cross-references to
link to named generator expressions.  Update some places to do this.

This is meant to demonstrate the capability, not as a comprehensive
sweep.

Commit:
AddFileDependencies: Avoid leaking temporary variable

Convert `add_file_dependencies` to a function.
Simplify the implementation using `set_property` with `APPEND`.

kwrobot committed:
CMake Nightly Date Stamp

kwrobot committed:
CMake Nightly Date Stamp

petitlapin committed:
CPack/NSIS: Add option for setting branding text

See https://nsis.sourceforge.io/Reference/BrandingText
for more information.

Fixes: #21479

kwrobot committed:
CMake Nightly Date Stamp

craigscott-crascit committed:
IOS_INSTALL_COMBINED: Support Xcode 12 (command line only)

Xcode 12 doesn't allow nested builds within the same build directory.
That means we can no longer do an install by building the install target
when IOS_INSTALL_COMBINED is true. We can, however, still do an install
by running the cmake_install.cmake script or executing cmake --install,
since there is no outer build and therefore the associated SDK can be
built as a sub-build.

The non-build methods previously didn't work when
IOS_INSTALL_COMBINED was true because the generated install script
and the CMakeIOSInstallCombined script both made certain assumptions
that relied on being part of a build. Those assumptions are now
removed. A side-effect of this work is that cpack now also works from the
command line when IOS_INSTALL_COMBINED is true.

Relates: #21282
Fixes: #20023

bradking committed:
Merge branch 'release-3.19'

bradking committed:
Merge topic 'lexer-posix-source' into release-3.19

a8b41e2c68 LexerParser: Do not override existing _POSIX_C_SOURCE definition
82cfeeb484 Merge branch 'backport-3.18-ci-wix-download' into release-3.18

Acked-by: Kitware Robot <kwrobot@kitware.com>
Merge-request: !5776

bradking committed:
Merge topic 'lexer-posix-source'

a8b41e2c68 LexerParser: Do not override existing _POSIX_C_SOURCE definition

Acked-by: Kitware Robot <kwrobot@kitware.com>
Merge-request: !5776

bradking committed:
Merge branch 'release-3.18'

bradking committed:
Merge topic 'parser-gcc-warning'

aac80ec1ee LexerParser: Suppress gcc -Wfree-nonheap-object warnings in generated code

Acked-by: Kitware Robot <kwrobot@kitware.com>
Acked-by: Christoph Grüninger <foss@grueninger.de>
Merge-request: !5775

bradking committed:
Merge topic 'doc-genex-links'

5dd356cefc Help: Convert some genex names to explicit cross-references

Acked-by: Kitware Robot <kwrobot@kitware.com>
Merge-request: !5778

bradking committed:
Merge topic 'FindGit-version-optimization-more'

c99dfd7be7 FindGit: Cache the version more effectively

Acked-by: Kitware Robot <kwrobot@kitware.com>
Merge-request: !5774

bradking committed:
Merge topic 'cmake_library_architecture_better_multiarch_support'

657fc3a9a7 CMakeDetermineCompilerABI: Parse library arch from versioned paths

Acked-by: Kitware Robot <kwrobot@kitware.com>
Merge-request: !5773

bradking committed:
AddFileDependencies: Deprecate this unnecessary module

Document that one can call `set_property` directly instead.

bradking committed:
Merge topic 'AddFileDependencies-simplify'

0f6fc388ea AddFileDependencies: Avoid leaking temporary variable

Acked-by: Kitware Robot <kwrobot@kitware.com>
Acked-by: Ben Boeckel <ben.boeckel@kitware.com>
Merge-request: !5779

bradking committed:
MacroAddFileDependencies: Explicitly deprecate by documentation

This module has been discouraged since commit 1e20be319f (STYLE: add
documentation for MACRO_ADD_FILE_DEPENDENCIES(), 2009-07-14,
v2.8.0~433).  It has been listed in the `cmake-modules(7)` manual
as deprecated since commit df780bcc01 (Help: Move deprecated modules to
appropriate section., 2018-11-12, v3.14.0-rc1~374^2).  Add the explicit
deprecation mark in its documentation.

bradking committed:
MacroAddFileDependencies: Simplify implementation

bradking committed:
Copyright.txt: Sort list of Contributors

bradking committed:
Merge topic 'copyright-sort'

046175b35a Copyright.txt: Sort list of Contributors

Acked-by: Kitware Robot <kwrobot@kitware.com>
Merge-request: !5789

Hsilgos committed:
FindXCTest: Fix output directory for test bundle with new build system

The output directory was still correct for macOS with the new build system,
but not for iOS.

Fixes: #20662

kwrobot committed:
CMake Nightly Date Stamp

bradking committed:
Merge topic 'AddFileDependencies-deprecate'

9cddaad940 AddFileDependencies: Deprecate this unnecessary module
4841d38a7a MacroAddFileDependencies: Explicitly deprecate by documentation
777c14af82 MacroAddFileDependencies: Simplify implementation

Acked-by: Kitware Robot <kwrobot@kitware.com>
Merge-request: !5788

bradking committed:
Merge topic '21479-branding-name-cpack-nsis'

0aabeb0c30 CPack/NSIS: Add option for setting branding text

Acked-by: Kitware Robot <kwrobot@kitware.com>
Merge-request: !5784

bradking committed:
Merge topic 'cuda_support_nvcc_from_hpctoolkit'

3cef91a321 CUDA: Always extract CUDA Toolkit root from nvcc verbose output

Acked-by: Kitware Robot <kwrobot@kitware.com>
Reviewed-by: Raul Tambre <raul@tambre.ee>
Tested-by: Raul Tambre <raul@tambre.ee>
Merge-request: !5777

bradking committed:
Merge topic 'object-lib-multiarch'

4d46b1401f add_library(): Allow imported object libraries with multi-arch
1a915e8953 Tests: TargetObjects does not need any languages enabled

Acked-by: Kitware Robot <kwrobot@kitware.com>
Merge-request: !5771

bradking committed:
Merge topic 'xcode12-ios_install_combined' into release-3.19

0110aa018d IOS_INSTALL_COMBINED: Support Xcode 12 (command line only)

Acked-by: Kitware Robot <kwrobot@kitware.com>
Merge-request: !5785

bradking committed:
Merge branch 'release-3.19'

bradking committed:
Merge topic 'xcode12-ios_install_combined'

0110aa018d IOS_INSTALL_COMBINED: Support Xcode 12 (command line only)

Acked-by: Kitware Robot <kwrobot@kitware.com>
Merge-request: !5785

bradking committed:
Merge branch 'release-3.19'

bradking committed:
Merge topic 'FindXCTest-bundle-inject' into release-3.19

eafe740ead FindXCTest: Fix output directory for test bundle with new build system

Acked-by: Kitware Robot <kwrobot@kitware.com>
Merge-request: !5786

bradking committed:
Merge topic 'FindXCTest-bundle-inject'

eafe740ead FindXCTest: Fix output directory for test bundle with new build system

Acked-by: Kitware Robot <kwrobot@kitware.com>
Merge-request: !5786

Luiz-Monad committed:
trycompile: option to specify shared_library

Luiz-Monad committed:
trycompile: target_variables

Luiz-Monad committed:
clr: wasn't setting CLRSupport tool config value

Luiz-Monad committed:
experimental

Luiz-Monad committed:
compiler C as CXX

bobbrow commented:
We're sorry that this PR was not addressed earlier.  Unfortunately, it is out of date at this point so our best course of action is to close it.

Luiz-Monad commented:
No problem, I wasn't supposed to make a pull request, I think it was a mistake.
